src/types/index.ts

```diff
@@ -208,3 +201,39 @@ export type SimulatedEventPayloadType = {
 
 export type ValueFunction<TValue, TArg> = (arg: TArg) => TValue;
 export type ValueOrFunction<TValue, TArg> = TValue | ValueFunction<TValue, TArg>;
+
+export interface OptionTypeBase {
+    [key: string]: any;
+}
+ type OptionsType<OptionType extends OptionTypeBase> = ReadonlyArray<OptionType>;
+
+ interface GroupTypeBase<OptionType extends OptionTypeBase> {
+    options: OptionsType<OptionType>;
+    [key: string]: any;
+}
```

Comment:
Is the indentation off here?

Reply:
Yeahhhh....there's some things here that's not great and challenging to work with. Tried to fix it up now